src/libs/Navigation/AppNavigator/AuthScreens.js

```diff
@@ -117,16 +124,22 @@ const defaultProps = {
     lastUpdateIDAppliedToClient: null,
 };
 
-class AuthScreens extends React.Component {
-    constructor(props) {
-        super(props);
+function AuthScreens({isSmallScreenWidth, isUsingMemoryOnlyKeys, lastUpdateIDAppliedToClient, session, lastOpenedPublicRoomID}) {
+    const screenOptions = getRootNavigatorScreenOptions(isSmallScreenWidth);
+    const isInitialRender = useRef(true);
 
+    if (isInitialRender.current) {
         Timing.start(CONST.TIMING.HOMEPAGE_INITIAL_RENDER);
+        isInitialRender.current = false;
```

Comment:
I think this should be inside of a `useEffect`, because `useEffect` is called when the render finished, in this case you are just checking the first time the code runs but not when the UI is done

Reply:
This is what I want to achieve.
In the class implementation, `Timing` is started in the constructor and stopped in componentDidMount.
https://github.com/Expensify/App/blob/main/src/libs/Navigation/AppNavigator/AuthScreens.js